src/App.jsx

```diff
@@ -4,50 +4,109 @@ import AddTodoForm from "./AddTodoForm"; //Import AddTodoForm component
-  const [isLoading, setIsLoading] = React.useState(true)
+  const [isLoading, setIsLoading] = React.useState(true);
 
-  React.useEffect(() => {
-    new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve({
-          data: {
-            todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
-          },
-        });
-      }, 2000);
-    }).then((result) => {
-      console.log("Fetched todo list:", result.data.todoList);
-      setTodoList(result.data.todoList);
+  async function fetchData() {
+    const url = `https://api.airtable.com/v0/${
+      import.meta.env.VITE_AIRTABLE_BASE_ID
+    }/${import.meta.env.VITE_TABLE_NAME}`;
+
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+      },
+    };
+
+    try {
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        const message = `Error: ${response.status}`;
+        throw new Error(message);
+      }
+
+      const data = await response.json();
+
+      const todos = data.records.map((todo) => {
+        return {
+          id: todo.id,
+          title: todo.fields.title,
+        };
+      });
+
+      setTodoList(todos);
       setIsLoading(false);
-    });
-  }, []);
+    } catch (error) {
+      console.log("Fetch error:", error.message);
+      return null
+    }
+  }
 
   React.useEffect(() => {
-    {/*early return to not run the code
-    if isLoading {
-      return;
-    }*/}
-if (!isLoading){
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-}}, [todoList, isLoading]);
-
-  const addTodo = (newTodo) => {
-    setTodoList([...todoList, newTodo]); //spread operator to add new todo to list
-  };
-
-  const removeTodo = (id) => {
-    const newList = todoList.filter((todo) => id !== todo.id);
-    setTodoList(newList);
-  };
-
-  return (
-    <>
-      <h1>Todo List</h1>
-      <AddTodoForm onAddTodo={addTodo} />
-      {isLoading ? (<p>"Loading..."</p>) : 
-      (<TodoList todoList={todoList} onRemoveTodo={removeTodo}/>)}
-    </>
-  );
-}
+    fetchData();
+  }, []);
+
+  async function addTodo(newTodoTitle) {
+
+    const url = `https://api.airtable.com/v0/${
+      import.meta.env.VITE_AIRTABLE_BASE_ID
+    }/${import.meta.env.VITE_TABLE_NAME}`;
+
+    const options = {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        records: [
+          {
+            fields: {
+              title: newTodoTitle,
+            },
+          },
+        ],
+      }),
+    };
+    
+    try {
+      const response = await fetch(url, options);
+      
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+
+      const data = await response.json();
+  
+      const newTodo = {
```

Comment:
try to console.log data, you have some issue in the newTodo

Reply:
Resolved! https://github.com/nihalerdal/react-todo/commit/f94100320e9b20c753b4226f12a04f19fe80b362